esphome/components/graphical_display_menu/graphical_display_menu.cpp

```diff
@@ -139,27 +139,32 @@ void GraphicalDisplayMenu::draw_menu_internal_(display::Display *display, const
   } else {
     if (scroll_menu_items) {
       // Attempt to draw the item after the current item (+1 for equality check in the draw loop)
-      last_item_index = std::min(this->cursor_index_ + 2, last_item_index);
+      last_item_index = std::min(last_item_index, this->cursor_index_ + 1);
 
       // Go back through the measurements to determine how many prior items we can fit
       int height_left_to_use = bounds->h;
-      for (int i = last_item_index - 1; i >= 0; i--) {
+      for (int i = last_item_index; i >= 0; i--) {
         display::Rect item_dimensions = menu_dimensions[i];
         height_left_to_use -= (item_dimensions.h + y_padding);
 
         if (height_left_to_use <= 0) {
           // Ran out of space -  this is our first item to draw
-          first_item_index = i + 1;
+          first_item_index = i;
           break;
         }
       }
+      // Dont't draw last item partially if it is the selected item
+      if (this->cursor_index_ == last_item_index && first_item_index < this->displayed_item_->items_size() - 1 &&
```

Comment:
```suggestion
      const int second_last_item = this->displayed_item_->items_size() - 1;
      const int number_items_being_drawn = last_item_index - first_item_index;
                      
      if (this->cursor_index_ == last_item_index && first_item_index < second_last_item && number_items_fit_to_screen <= number_items_being_drawn) {
```

Could we look to simplify this expression?

Reply:
Of course. I already tried to make it more readable by splitting it over multiple lines, but the all-knowing linter didn't allow it.

Your suggestion adds back what I tried to remove: mixing indices with item counter:
`const int second_last_item = this->displayed_item_->items_size() - 1;` is actually the last item.

Reply:
Yeah, I don't love some of the lint settings used. Other option would be to throw some brackets around it to help make the subexpressions clearer - but not sure if the linter would like that, off the top of my head.